src/plots/barPlot.ts

```diff
@@ -420,68 +420,78 @@ export module Plots {
+
+        let getY = () => {
+          let addend = 0;
+          if (this._isVertical)  {
+              addend += offset;
+              if (showLabelOffBar && positive) {
+                  addend += (offset - h);
+              }
+              if (showLabelOffBar && !positive) {
+                  addend += measurement.height;
+              };
+          }
+          return baseY + addend;
+        };
+
+        let getX = () => {
+          let addend = 0;
+          if (!this._isVertical)  {
+              addend += offset;
+              if (showLabelOffBar && positive) {
+                  addend += (offset - w - Bar._LABEL_HORIZONTAL_PADDING);
+              }
+              if (showLabelOffBar && !positive) {
+                  addend += measurement.width;
+              };
           }
+          return baseX + addend;
+        };
 
-          let showLabel = true;
-          let labelPosition = {
-            x: x,
-            y: positive ? y : y + h - measurement.height
-          };
+        let x = getX();
+        let y = getY();
 
-          if (this._isVertical) {
-            labelPosition.x = x + w / 2 - measurement.width / 2;
-          } else {
-            if (!positive) {
-              labelPosition.x = x + w - measurement.width;
-            } else {
-              labelPosition.x = x;
-            }
-          }
+        let g = labelArea.append("g").attr("transform", "translate(" + x + "," + y + ")");
 
-          if (labelPosition.x < 0 || labelPosition.x + measurement.width > this.width() ||
-              labelPosition.y < 0 || labelPosition.y + measurement.height > this.height()) {
-            showLabel = false;
-          }
+        let labelPositioningClassName = showLabelOffBar ? "off-bar-label" : "on-bar-label";
+        g.classed(labelPositioningClassName, true);
 
-          let g = labelArea.append("g").attr("transform", "translate(" + x + "," + y + ")");
-          let className = dark ? "dark-label" : "light-label";
-          g.classed(className, true);
-          g.style("visibility", showLabel ? "inherit" : "hidden");
-          let xAlign: string;
-          let yAlign: string;
-          if (this._isVertical) {
-            xAlign = "center";
-            yAlign = positive ? "top" : "bottom";
-          } else {
-            xAlign = positive ? "left" : "right";
-            yAlign = "center";
-          }
-          let writeOptions = {
-              selection: g,
-              xAlign: xAlign,
-              yAlign: yAlign,
-              textRotation: 0
-          };
-          writer.write(text, w, h, writeOptions);
+        let color = attrToProjector["fill"](d, i, dataset);
+        let dark = Utils.Color.contrast("white", color) * 1.6 < Utils.Color.contrast("black", color);
+        g.classed(dark ? "dark-label" : "light-label", true);
+
+        let hideLabel = (x + measurement.width > this.width() || (positive ? y + measurement.height : y + h) > this.height());
```

Comment:
This isn't quite the right calculation, since half the bar might have to be added, depending on the orientation of the `Plot`. The original code took that into account:
```Typescript
let showLabel = true;
let labelPosition = {
  x: x,
  y: positive ? y : y + h - measurement.height
};

if (this._isVertical) {
  labelPosition.x = x + w / 2 - measurement.width / 2;
} else {
  if (!positive) {
    labelPosition.x = x + w - measurement.width;
  } else {
    labelPosition.x = x;
  }
}

if (labelPosition.x < 0 || labelPosition.x + measurement.width > this.width() ||
    labelPosition.y < 0 || labelPosition.y + measurement.height > this.height()) {
  showLabel = false;
}
```